src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -95,7 +95,13 @@ struct module_param {
 	 * Specifies the unique id of a parameter. For example the parameter
 	 * sample_rate may have an id of 0x01.
 	 */
-	uint32_t id;
+	uint16_t id;
+	/**
+	 * Specifies the codec id for which this parameter id applies. This
+	 * is useful when supporting multiple codecs. If single codec is
+	 * used this field can be set to zero.
+	 */
+	uint16_t codec_id;
```

Comment:
@dbaluta since the module interface is used by all kinds of components and not just for compressed codec implementation, Im not sure if codec_id in the common struct makes sense here. Maybe we should leave this as ID and have each codec handle the encoding on codec_id and param_id with a codec-specific implementation that the common module interface can be oblivious to. something like this for cadence codecs for example:

uint16_t params_id = params->id && 0xFF;
uint16_t codec_id = params->id >> 16;

Reply:
@ranj063 I think this is a good suggestion. will update the PR.